File: tests/builder4/Unit1.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "Unit1.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "rpcompobase"
#pragma link "rpvclreport"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TForm1::Button1Click(TObject *Sender)
{
 VCLReport1->Filename=Edit1->Text;
 VCLReport1->Execute();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button2Click(TObject *Sender)
{
 PrinterSetupDialog1->Execute();        
}
//---------------------------------------------------------------------------

File: tests/clxreport/Unit1.pas
unit Unit1;

interface

uses
  SysUtils, Types, Classes, QGraphics, QControls, QForms, QDialogs,
  QStdCtrls, rpclxreport,  DB,
{$IFDEF MSWINDOWS}
  rpvclreport,
{$ENDIF}
  rpcompobase, rpmdesigner;


type
  TForm1 = class(TForm)
    CLXReport1: TCLXReport;
    Button1: TButton;
    Button2: TButton;
    Label1: TLabel;
    ETest: TEdit;
    RpDesigner1: TRpDesigner;
    Button3: TButton;
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
    procedure Button3Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.xfm}

procedure TForm1.Button1Click(Sender: TObject);
begin

 CLXReport1.Execute;
end;

procedure TForm1.Button2Click(Sender: TObject);
begin
 CLXReport1.SaveToPDF(ETest.text);
end;

procedure TForm1.Button3Click(Sender: TObject);
begin
 RpDesigner1.Execute;
end;

end.
